src/test/collectionHandlingFunctions.php

```diff
@@ -0,0 +1,314 @@
+                        "image" => NULL      
+                    ];
+        $expectedOutput = '<div class="collection_item">';
+        $expectedOutput .= '<h3 tabindex="0">Unicorn brand Rainbow</h3>';
+        
+        //When image is undefined a stock image is used 
+        $expectedOutput .= '<img tabindex="0" src="images/colorful-wool.jpg" alt="Stock image of colorful yarn">';              
+
+        //this is unchanged
+        $expectedOutput .= '<h4 tabindex="0">Description</h4>';
+        $expectedOutput .= '<ul>';
+        $expectedOutput .= '<li>Dominant colour: purple</li>';
+        $expectedOutput .= '<li>Yarn type: Super Chunky</li>';
+        $expectedOutput .= '<li>Composition: Cashmere</li>';
+        $expectedOutput .= '<li>Length of a skein: 100 m</li>';            
+        $expectedOutput .= '<li>Quantity: 2 skeins in stash</li>';
+        $expectedOutput .= '<li>Allocated to project: Cool rainbow hat</li>';
+        $expectedOutput .= '</ul>';
+        $expectedOutput .= '</div>'; 
+        
+        $actualOutput = describeCollectionItem($yarnExample);
+
+        $this->assertEquals($expectedOutput, $actualOutput);
+    }
+
+    public function testSuccessDescribeCollectionItem_undefinedProject()
+    {
+        $yarnExample = ["name" => "Unicorn brand", 
+                        "shade" => "Rainbow", 
+                        "dominantColour" => "purple",
+                        "composition" => "Cashmere",
+                        "yarnType" => "Super Chunky", 
+                        "lengthInMeters" => 100,
+                        "skeinNumber" => 2, 
+                        "project"=> NULL,
+                        "image" => "blue_scarf.jpg"      
+                    ];
+        // this doesn't change            
+        $expectedOutput = '<div class="collection_item">';
+        $expectedOutput .= '<h3 tabindex="0">Unicorn brand Rainbow</h3>';
+        $expectedOutput .= '<img tabindex="0" src="images/blue_scarf.jpg" alt="A picture of Unicorn brand in shade Rainbow">';              
+        $expectedOutput .= '<h4 tabindex="0">Description</h4>';
+        $expectedOutput .= '<ul>';
+        $expectedOutput .= '<li>Dominant colour: purple</li>';
+        $expectedOutput .= '<li>Yarn type: Super Chunky</li>';
+        $expectedOutput .= '<li>Composition: Cashmere</li>';
+        $expectedOutput .= '<li>Length of a skein: 100 m</li>';            
+        $expectedOutput .= '<li>Quantity: 2 skeins in stash</li>';
+
+        // the following line is printed when project is undefined
+        $expectedOutput .= '<li>Available for a new project!</li>';
+        $expectedOutput .= '</ul>';
+        $expectedOutput .= '</div>'; 
+        
+        $actualOutput = describeCollectionItem($yarnExample);
+
+        $this->assertEquals($expectedOutput, $actualOutput);
+    }
+
+// in the following functions, we check that the bullet point of the missing property is not printed
```

Comment:
isn't this comment basically the same as the one on line 61?

Reply:
Not totally: for these functions, I check that a missing field results in a missing bullet point,  whereas in the previous functions I would change the some other parts of the text (e.g. the content of `<img>` tag). But maybe this comment is not that useful? I removed it.